### DISCUSSION TITLE
Basic PDF support

### DISCUSSION THREAD

**Discussion by pxlrbt** [pxlrbt/filament-excel#258] [Q&A]
From Discord:

> Hi Dennis, I hope you're doing well. I just wanted to let you know that I was able to landscape the PDF using the event as you suggested. Thank you for your help!
> Here are the code I have used:

```php
Sheet::listen(BeforeSheet::class, function ($sheet) {
            $sheet->getDelegate()->getPageSetup()->setOrientation(\PhpOffice\PhpSpreadsheet\Worksheet\PageSetup::ORIENTATION_LANDSCAPE);

            $sheet->getDelegate()->getStyle('1')->getFont()->setSize(8);
        });
```

> However, you can use the RegistersEventListeners trait to automatically register the event listeners. To do so, simply add the trait to your class and include this function within your class:

```php
beforeSheet(BeforeSheet $event)
    {
         $event->sheet->getDelegate()->getPageSetup()->setOrientation(\PhpOffice\PhpSpreadsheet\Worksheet\PageSetup::ORIENTATION_LANDSCAPE);
    }
```

> or you can just add this one in your Page without trait

```php
public function boot()
    {
        Sheet::listen(BeforeSheet::class, function ($sheet) {
            $sheet->getDelegate()->getPageSetup()->setOrientation(\PhpOffice\PhpSpreadsheet\Worksheet\PageSetup::ORIENTATION_LANDSCAPE);

            $sheet->getDelegate()->getStyle('1')->getFont()->setSize(8);
        });
    }
```

**Comment by jimiero** [comment]
@pxlrbt i'm using this on my resource table

```php
->headerActions([
                ExportAction::make('pdf')
                    ->label('PDF')
                    ->color('gray')
                    ->exports([
                        ExcelExport::make()->withWriterType(\Maatwebsite\Excel\Excel::TCPDF)->label('PDF')->fromTable()->withFilename('my-resource.pdf'),
                    ])
            ])
```

how can i change that to provide the pdf orientation when it exports to pdf my table?

**Comment by pxlrbt** [comment]
Did you try the approach mentioned above?
I never exported to PDF via Maatswebsite\Excel package, so I can't tell more.

**Comment by jimiero** [comment]
I don't know where exactly to use the code above, and I'm not using a custom export as you see my action above

**Comment by pxlrbt** [comment]
Either to a ServiceProvider (first example) or your page class (third example)

**Comment by jimiero** [comment]
thanks, works using ServiceProvider, but now i have another issues 🤦  the table column widths get too big and the pdf doesn't look fine.. any idea if I can shrink those inside the pdf?

**Comment by pxlrbt** [comment]
Sorry, as I said, I never used the PDF export before, so I can't really help with this.
